src/mds/MDSAuthCaps.cc

```diff
@@ -61,16 +62,20 @@ struct MDSCapParser : qi::grammar<Iterator, MDSAuthCaps()>
       lexeme[lit("'") >> *(char_ - '\'') >> '\''];
     unquoted_path %= +char_("a-zA-Z0-9_./-");
     network_str %= +char_("/.:a-fA-F0-9][");
+    fs_name_str %= +char_("a-zA-Z0-9-_.");
 
     // match := [path=<path>] [uid=<uid> [gids=<gid>[,<gid>...]]
     path %= (spaces >> lit("path") >> lit('=') >> (quoted_path | unquoted_path));
     uid %= (spaces >> lit("uid") >> lit('=') >> uint_);
     uintlist %= (uint_ % lit(','));
     gidlist %= -(spaces >> lit("gids") >> lit('=') >> uintlist);
+    fs_name %= -(spaces >> lit("fsname") >> lit('=') >> fs_name_str);
     match = -(
 	     (uid >> gidlist)[_val = phoenix::construct<MDSCapMatch>(_1, _2)] |
 	     (path >> uid >> gidlist)[_val = phoenix::construct<MDSCapMatch>(_1, _2, _3)] |
-             (path)[_val = phoenix::construct<MDSCapMatch>(_1)]);
+             (fs_name >> path)[_val = phoenix::construct<MDSCapMatch>(_2, _1)] |
+             (path)[_val = phoenix::construct<MDSCapMatch>(_1)] |
+             (fs_name)[_val = phoenix::construct<MDSCapMatch>(_1, 0)]);
```

Comment:
```suggest
(fs_name)[_val = phoenix::construct<MDSCapMatch>(std::string(),  _1)]);
```
If you make this change you can get rid of the following constructor in MDSAuthCaps.h
```
explicit MDSCapMatch(std::string fs_name, int i)
```

Reply:
Yes, that's much better/. Thanks, I've incorporated the change.